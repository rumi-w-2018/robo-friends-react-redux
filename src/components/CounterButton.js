import React, { Component } from 'react';

class CounterButton extends Component {
  constructor(props) {
    super(props);

    this.state = {
      count: 1
    };
  }

  // Re-render only when the state changes
  shouldComponentUpdate(nextProps, nextState) {
    //console.log(nextProps, nextState);
    if (this.state.count !== nextState.count) {
      return true;
    }
    return false;
  }

  updateCount = () => {
    this.setState(state => {
      return {
        count: this.state.count + 1
      };
    });
  };

  render() {
    //console.log('counter');
    return (
      <button id="counter" color={this.props.color} onClick={this.updateCount}>
        Count: {this.state.count}
      </button>
    );
  }
}

export default CounterButton;
