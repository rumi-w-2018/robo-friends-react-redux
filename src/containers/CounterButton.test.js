import { shallow } from 'enzyme';
import React from 'react';
import CounterButton from './CounterButton';

it('exptects to render CounterButton component', () => {
    const mockColor = 'yellow';
    expect(shallow(<CounterButton color={mockColor} />)).toMatchSnapshot();
});

it('correctly increments the cunter', () => {
    const mockColor = 'yellow';
    const wrapper = shallow(<CounterButton color={mockColor} />);

    wrapper.find('[id="counter"]').simulate('click');
    expect(wrapper.state()).toEqual({ count:2 });

    wrapper.find('[id="counter"]').simulate('click');
    expect(wrapper.state()).toEqual({ count:3 });
    //console.log(wrapper.props());
    expect(wrapper.props().color).toEqual('yellow');

});