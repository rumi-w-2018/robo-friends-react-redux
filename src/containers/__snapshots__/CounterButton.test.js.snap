// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`exptects to render CounterButton component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CounterButton
    color="yellow"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Count: ",
        1,
      ],
      "color": "yellow",
      "id": "counter",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      "Count: ",
      1,
    ],
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Count: ",
          1,
        ],
        "color": "yellow",
        "id": "counter",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        "Count: ",
        1,
      ],
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
